### PR TITLE
fix: Fix S3 file loading

### PR DESCRIPTION
**Pull request by simster7**
Fixes: https://github.com/argoproj/argo-workflows/issues/5352

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/argoproj/argo-workflows/v3
 
 go 1.15
 
+replace github.com/argoproj/pkg v0.7.0 => github.com/simster7/pkg v0.2.1-0.20210310152523-7adcfcfad809
```
Comment (simster7) [inline go.mod:5]: References https://github.com/argoproj/pkg/pull/38

File: workflow/artifacts/s3/s3.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func (s3Driver *ArtifactDriver) ListObjects(artifact *wfv1.Artifact) ([]string,
 			if err != nil {
 				return false, err
 			}
-			files, err = s3cli.ListDirectory(artifact.S3.Bucket, artifact.S3.Key)
+			files, err = s3cli.ListDirectory(artifact.S3.Bucket, "")
```
Comment (alexec) [inline workflow/artifacts/s3/s3.go:149]: why remove key?

Comment (simster7) [reply workflow/artifacts/s3/s3.go:149]: Oversight here

File: go.mod [hunk 3]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/aliyun/aliyun-oss-go-sdk v2.1.5+incompatible
 	github.com/antonmedv/expr v1.8.8
 	github.com/argoproj/argo-events v1.2.0
-	github.com/argoproj/pkg v0.7.0
+	github.com/argoproj/pkg v0.8.0
```
[the remark this answers is not in the document]
Comment (terrytangyuan) [reply go.mod:16]: Just curious for future reference when releasing small fixes

**Review by alexec — APPROVED** [review]

**Pull request merged**